framework/transport/ieee1905_transport/ieee1905_transport_broker.cpp

```diff
@@ -0,0 +1,368 @@
+/////////////////////////// Local Module Functions ///////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+static bool is_restricted_type(uint32_t type)
+{
+    // TODO: Until VS message filtering is implemented (based on action and action_op)
+    //       assume there are no restricted types. After the implementation,
+    //       subscribing to BROKER specific message should be restricted.
+
+    return false;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////// Broker Implementation ////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+BrokerServer::BrokerServer(const std::string &broker_uds_path, SocketEventLoop::TimeoutType timeout)
+    : SocketEventLoop(timeout)
+{
+    m_broker_uds_path = broker_uds_path;
+
+    // Server sockets event handlers
+    EventHandlers server_socket_handlers{
+        // Accept incoming connections
+        .on_read =
+            [&](EventType socket, EventLoopType &loop) {
+                // cppcheck falsely detects the call to socket_connected() as a call to a
+                // virtual method from whithin the constructor, so suppress this warning,
+                // by explicitly calling BrokerServer::socket_connected().
+
+                // Handle connections to one of the server sockets
+                if (!BrokerServer::socket_connected(
+                        std::dynamic_pointer_cast<SocketServer>(socket))) {
+                    // NOTE: Do NOT stop the broker on connection errors...
+                }
+
+                return true;
+            },
+
+        // Not implemented
+        .on_write   = nullptr,
+        .on_timeout = nullptr,
+
+        // Fail on server socket disconnections or errors
+        .on_disconnect =
+            [&](EventType socket, EventLoopType &loop) {
+                LOG(ERROR) << "Broker socket disconnected!";
+                return false;
+            },
+        .on_error =
+            [&](EventType socket, EventLoopType &loop) {
+                LOG(ERROR) << "Broker socket error!";
+                return false;
+            },
+    };
+
+    // Connect to the message broker
+    m_broker_socket_uds = std::make_shared<SocketServer>(m_broker_uds_path, listen_buffer_size);
+    LOG_IF(!m_broker_socket_uds, FATAL) << "Failed allocating memory!";
+
+    // Check for errors
+    LOG_IF(!m_broker_socket_uds->getError().empty(), FATAL)
+        << "Failed opening UDS socket: " << m_broker_uds_path
+        << " [ERROR: " << m_broker_socket_uds->getError() << "]";
+
+    // Add the socket to the poll
+    LOG_IF(!add_event(m_broker_socket_uds, server_socket_handlers), FATAL)
+        << "Failed adding the broker socket into the poll";
+
+    // Create the TCP server socket
+    m_broker_socket_tcp = std::make_shared<SocketServer>(BROKER_TCP_PORT, listen_buffer_size);
```

Comment:
I don't get it - why do we need a TCP server socket? Each device should have one broker running which "connects" the controller and agent to the transport. The transport sends the actual CMDUs on the raw socket / bridge, and sends messages received from outside to the broker which then forwards them to the subscribers. What am I missing?

Reply:
We don't really need it, I thought to that it may be useful in the future to support "external" applications (e.g. Mobile apps replacing the `beerocks_analyzer`), but they will probably use the high-level API anyway, so I'll remove this.